src/Development/Shake/Internal/Derived.hs

```diff
@@ -72,22 +75,30 @@ addShakeExtra x = Map.insert (typeOf x) (toDyn x)
 
 -- | @copyFile' old new@ copies the existing file from @old@ to @new@.
 --   The @old@ file will be tracked as a dependency.
+--   Also creates the new directory if necessary.
 copyFile' :: FilePath -> FilePath -> Action ()
 copyFile' old new = do
     need [old]
     putLoud $ "Copying from " ++ old ++ " to " ++ new
+    createDirIfMissing new
     liftIO $ copyFile old new
 
+createDirIfMissing :: FilePath -> Action()
+createDirIfMissing new =
+   liftIO $ createDirectoryIfMissing True $
+          (if isWindows then FW.takeDirectory else FP.takeDirectory) new
```

Comment:
No need for the isWindows test - if you just use System.FilePath.takeDirectory then it is basically an alias for what you wrote and will work on both platforms.